tsec-http4s/src/main/scala/tsec/authentication/SecuredRequestHandler.scala

```diff
@@ -9,20 +10,20 @@ import tsec.authorization._
 
 sealed abstract class SecuredRequestHandler[F[_], Identity, User, Auth](
     val authenticator: Authenticator[F, Identity, User, Auth]
-)(implicit F: MonadError[F, Throwable]) {
+)(implicit F: MonadError[F, Throwable], AE: ApplicativeError[Kleisli[OptionT[F, ?], Request[F], ?], Throwable]) {
```

Comment:
Implicit needed here because they changed something in cats so the the `ApplicativeError[F, Unit]` takes highest priority 👎

Reply:
Hm. 

Well, I made it `OptionT`, at the time, because our `HttpRoutes` were `OptionT`, and it just seemed to make sense.

To be honest, a real solution is:
- Enumerate the errors into an `AuthError` adt (that extends `NoStackTrace` and `Exception`
- `.recover` on specific sections of the error composition and re-raise (for efficiency)
- return `EitherT` by recovering over the whole stmt.

It's a possible solution, anyway.

Reply:
I think the argument that `OptionT` is correct because it is the same type as `HttpRoutes` is still valid.

Made a quick change to the code to make why I added this more clear. Here is what I posted in the http4s channel to see if this was intentional, because cats had a breaking change:

> doing a port of tsec to 0.19 and noticed a change that (imo) reduces some ergonomics.
> we used to be able to do (svc: HttpService[F]).handleErrorWith(t: Throwable => ...)
> so i expected (routes: HttpRoutes[F]).handleErrorWith(t: Throwable => ...) would work the same
> but it seems the provided implicit expects a Unit
> eg (routes: HttpRoutes[F]).handleErrorWith(u: Unit => ...)
> i think this was because of a cats change?
> anyway, i think i can get around this with an implicit ME: MonadError[Kleisli[OptionT[F, ?], Request[F], ?], Throwable]
> and then ME.handleErrorWith(routes: HttpRoutes[F])(t: Throwable => ...)
> but that feels kinda meh.

I'm ok, leaving that as-is. What do you think?

Reply:
Also going to ask if they can put this implicit in http4s so we won't need it here.